golem-worker-executor-base/src/durable_host/mod.rs

```diff
@@ -312,19 +312,22 @@ impl<Ctx: WorkerCtx> DurableWorkerCtx<Ctx> {
         WasiHttpImpl(DurableWorkerCtxWasiHttpView(self))
     }
 
-    pub async fn update_worker_status(&self, f: impl FnOnce(&mut WorkerStatusRecord)) {
-        let mut status = self
-            .execution_status
+    pub fn get_worker_status_record(&self) -> WorkerStatusRecord {
+        self.execution_status
             .read()
             .unwrap()
             .last_known_status()
-            .clone();
+            .clone()
+    }
 
-        let mut deleted_regions = self.state.replay_state.deleted_regions().await;
+    pub async fn update_worker_status(&self, f: impl FnOnce(&mut WorkerStatusRecord)) {
+        let mut status = self.get_worker_status_record();
+
+        let mut skipped_regions = self.state.replay_state.skipped_regions().await;
         let (pending_updates, extra_deleted_regions) = self.public_state.worker().pending_updates();
```

Comment:
```suggestion
        let (pending_updates, extra_skipped_regions) = self.public_state.worker().pending_updates();
```
If I understand the new naming correctly